src/guards/role.guard.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  CanActivate,
+  Injectable,
+  ExecutionContext,
+  UnauthorizedException,
+  Logger,
+  Inject,
+} from '@nestjs/common'
+
+import { KeycloakService } from '../service'
+import { Reflector } from '@nestjs/core'
+import { META_PUBLIC } from '../decorators/public.decorator'
+import { META_ROLES, RoleDecoratorOption } from '../decorators/roles.decorator'
+import { RoleMatchingMode } from '../constants'
+import KeycloakConnect from 'keycloak-connect'
+
+@Injectable()
+export class Roleuard implements CanActivate {
+  logger = new Logger(Roleuard.name)
+
+  constructor(
+    @Inject(KeycloakService)
+    private keycloak: KeycloakService,
+    private readonly reflector: Reflector
+  ) {}
+
+  getRequest(context: ExecutionContext): any {
+    return context.switchToHttp().getRequest()
+  }
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const isPublic = this.reflector.get<boolean>(META_PUBLIC, context.getHandler())
+
+    // Emit process, if endpoint is public.
+    if (isPublic) {
+      return true
+    }
+
+    const request = this.getRequest(context)
+
+    const roleDefinition = this.reflector.get<RoleDecoratorOption>(META_ROLES, context.getHandler())
+
+    // If no resource type is defined as class decorator, emit.
+    if (!roleDefinition || !roleDefinition?.roles?.length) {
```

Comment:
We don't need `!roleDefinition` since `!roleDefinition?.roles` already handles the case there roleDefinition is not defined

Reply:
Yeah you are right, i'll fix it